tests/func/metrics/test_show.py

```diff
@@ -48,3 +51,34 @@ def test_show_branch(tmp_dir, scm, dvc):
         "working tree": {"metrics.yaml": {"foo": 1}},
         "branch": {"metrics.yaml": {"foo": 2}},
     }
+
+
+def test_show_subrepo_with_preexisting_tags(tmp_dir, scm):
+    tmp_dir.gen("foo", "foo")
+    scm.add("foo")
+    scm.commit("init")
+    scm.tag("no-metrics")
+
+    tmp_dir.gen({"subdir": {}})
+    subrepo_dir = tmp_dir / "subdir"
+    with subrepo_dir.chdir():
+        dvc = Repo.init(subdir=True)
+        scm.commit("init dvc")
+
+        subrepo_dir.gen("metrics.yaml", "foo: 1")
+        dvc.run(metrics=["metrics.yaml"], single_stage=True)
```

Comment:
Needs command, and please rebase as https://github.com/iterative/dvc/pull/3850 has been merged.